### PR TITLE
feat: check run output gas_used

### PR DESCRIPTION
**Pull request by enitrat**
# Pull Request type

Time spent on this PR: 0.3d

Resolves: #{Issue number}

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## What is the new behavior?

Uses the output from the run to retrieve the `gas_used` value, instead of using the value from the block header.

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: crates/ef-testing/src/models/result.rs [hunk 1]
```diff
@@ -5,75 +5,108 @@ use blockifier::{
         objects::{TransactionExecutionInfo, TransactionExecutionResult},
     },
 };
+use eyre::{eyre, Result};
```
Comment (greged93) [inline crates/ef-testing/src/models/result.rs:8]: Can we use cainome for this ?

File: crates/ef-testing/src/models/case.rs [hunk 2]
```diff
@@ -97,35 +97,37 @@ impl BlockchainTestCase {
...
+        });
 
         // Get gas_used and base_fee from RLP block - as in some cases, the block header is not present in the test data.
         let sealed_block = SealedBlock::decode(&mut self.block.rlp.as_ref())
             .map_err(RunnerError::RlpDecodeError)?;
         let sealed_header = sealed_block.header.unseal();
 
         let coinbase = sealed_header.beneficiary;
-        let gas_used: U256 = U256::from(sealed_header.gas_used);
+        let gas_used: U256 = U256::from(output.gas_used);
```
Comment (greged93) [inline crates/ef-testing/src/models/case.rs:130]: I'm not sure why we do this: I think we should compare the two values, not just replace one with the other

Comment (enitrat) [reply crates/ef-testing/src/models/case.rs:130]: Used `expected` and `actual` values, compared them and pushed in `diff` message if not coherent.

**Review by greged93 — APPROVED** [review]

**Review by greged93 — COMMENTED** [review]
just checking

**Pull request merged**